Add non-skinnable position indicator to multiplayer player

bdach opened the pull request:
One of the [stated constraints](https://github.com/ppy/osu/issues/27850#issuecomment-2556485304) in making the gameplay leaderboard toggleable / skinnable / deletable was that there needs to be *something* local to multiplayer that cannot be deleted to distinguish it from offline play. This introduces such an element, visually similar to the position display shown on the lower right side of the screen on stable. This is how it looks like:

https://github.com/user-attachments/assets/2d815423-aaa8-47f5-a9da-021ebdcd4bc2

The logic here will match stable - pressing <kbd>Tab</kbd> will hide this display during gameplay, but it *will* always show during breaks regardless of toggle states.

The visuals of the display are *somewhat* muted, because the most common reason cited for not wanting the leaderboard to display in multiplayer is that it's "distracting", so the logic here is that if the display is subtle and does not try to animate all over, then maybe it won't be distracting.

(Or maybe the logic is that the distraction is in being able to see if you're throwing the match or not. If that is the argument then the reasoning doesn't work, and the only thing that works is *not seeing that you're throwing the match*. We'll see what the user reception of this is going to be.)

---

- [x] Depends on https://github.com/ppy/osu/pull/32939
- [x] Depends on https://github.com/ppy/osu/pull/32947

Cross-reference for correspondent stable logic:

- https://github.com/peppy/osu-stable-reference/blob/996648fba06baf4e7d2e0b248959399444017895/osu!/GameModes/Play/PlayerVs.cs#L172-L176
- https://github.com/peppy/osu-stable-reference/blob/996648fba06baf4e7d2e0b248959399444017895/osu!/GameModes/Play/Player.cs#L3482-L3495

bdach committed:
Implement local user position display for multiplayer

bdach committed:
Merge branch 'use-actual-score-positions-in-gameplay-leaderboard' into multiplayer-position-indicator

bdach committed:
Add local user position display to multiplayer

smoogipoo commented:
I didn't mention this at first (not something to consider for this PR, just stating my overall position) but I thought it'd be cool to have a Mario Kart-esque counter, with the whole gold/silver/bronze/everything-else grading system.

Basically what I'm saying is persistent _tasteful_ distractions could be nice to have, as opposed to no/minimal distractions.

bdach commented:
> I thought it'd be cool to have a Mario Kart-esque counter, with the whole gold/silver/bronze/everything-else grading system.
> 
> Basically what I'm saying is persistent _tasteful_ distractions could be nice to have

I don't feel comfortable making such stylistic calls myself for at least two reasons (I dunno what @peppy or the tournament player clique would say to that).

peppy commented:
I agree with smoogi on this one, and have plans to give throw some creativity at this PR before merging. I have a few ideas in my head which involves adding a small line/bar area beneath or around the number that gives an overview of the room without taking up too much screen / eye real estate.

bdach committed:
Merge branch 'master' into multiplayer-position-indicator

peppy committed:
Fix test not showing immediately useful state

Also standardises the testing leaderboard provider component for
(immediate) future use.

bdach commented:
I did not intend to set it up for multiplayer spectator. I'm not particularly opposed to it but I am somewhat wary of information duplication and maybe also visibility concerns (wouldn't want the indicators to take too much room on already smaller-size player instances).

peppy committed:
Add bar display

peppy committed:
Improve transition fade in/out

peppy committed:
Roll the rank counter instead of immediate updates

bdach committed:
Fix code inspection

bdach committed:
Fix multiplayer position display not hiding on user change

bdach commented:
I don't have any particular objections, we'll see what user reception ends up being.
